File: php/src/AssessService/Assessor.php
<?php

declare(strict_types=1);

namespace FactChecker\AssessService;

interface Assessor
{
    const DEFAULT_SCORE = 0;
    const MAX_SCORE = 5;

    const DEFAULT_OPINIONS = [
        'unassessable',
        'unreliable',
        'plausible',
        'believable',
        'credible',
    ];

    /**
     * Calculate a score of a sentence.
     *
     * @param string $sentence
     * @return int
     */
    public function getScore(string $sentence): int;

    /**
     * Form an opinion about a sentence.
     *
     * @param string $sentence
     * @return string
     */
    public function getOpinion(string $sentence): string;
}

File: java/app/src/main/java/factchecker/AssessService/Assessor.java
package factchecker.AssessService;

public interface Assessor {
    public static final Integer DEFAULT_SCORE = 0;
    public static final Integer MAX_SCORE = 5;

    public static final String[] DEFAULT_OPINIONS = {
        "unassessable",
        "unreliable",
        "plausible",
        "believable",
        "credible",
    };

    /**
     * Calculate a score for a sentence.
     */
    abstract Integer getScore(String sentence);

    /**
     * Form an opinion about a sentence.
     */
    abstract String getOpinion(String sentence);
}
